upgrade_tests/helpers/existence.py

```diff
@@ -103,6 +104,63 @@
 attribute_keys['content-view'] = 'content view id'
 
 
+def _find_on_list_of_dicts(lst, data_key, all=False):
+    """Returns the value of a particular key in a dictionary from the list of
+    dictionaries, when 'all' is set to false.
+
+    When 'all' is set to true, returns the list of values of given key from all
+    the dictionaries in list.
+
+    :param list lst: A list of dictionaries
+    :param str data_key: A key name of which data to be retrieved from given
+        list of dictionaries
+    :param bool all: Fetches all the values of key in list of dictionaries if
+        True, else Fetches only single and first value of a key in list of
+        dictionaries
+    :returns the list of values or a value of a given data_key depends on
+        'all' parameter
+
+    """
+    dct_values = [dct.get(data_key) for dct in lst]
+    if all:
+        return dct_values
+    for v in dct_values:
+        if v is not None:
+            return v
+
+    raise KeyError(
+        'Unable to find data for key \'{0}\' in satellite.'.format(
+            data_key))
+
+
+def _find_on_list_of_dicts_using_search_key(lst, search_key, attr):
```

Comment:
```python
dict1 = {
    'age': '12', 'address': '34 Main Street, 212 First Avenue',
    'firstName': 'Alan', 'lastName': 'Stone'
}

dict2 = {
    'age': '13', 'address': '22 Acacia Avenue',
    'firstName': 'Mary-Ann', 'lastName': 'Lee'
}

dict3 = {
    'age': '13', 'address': '22 Acacia Avenue',
    'firstName': 'Mary-Jane', 'lastName': 'Lee'
}


def search(list_of_dicts, term, key):
    """Search for given `term` in the `list of dicts`
    only if value is found under `key`
    """
    for single_dict in list_of_dicts:
        for k, v in single_dict.items():
            if term in v and k == key:
                return v


# Checking if string 'Mary' exists in dictionary value under key `firstName`
# prints Mary-Ann
# Mary-Jane is never printed because the order of dicts
print(search([dict1, dict2, dict3], 'Mary', key='firstName'))

```